packages/aws-cdk-lib/aws-ec2/lib/machine-image/machine-image.ts

```diff
@@ -156,6 +156,21 @@ export abstract class MachineImage {
     return new GenericSsmParameterImage(parameterName, options);
   }
 
+  /**
+   * An image specified in SSM parameter store that will be resolved at instance launch time.
+   *
+   * The AMI ID will be resolved at instance launch time.
+   *
+   * @param parameterName The name of SSM parameter containing the AMI ID
+   * @param options The parameter image options
+   *
+   * @see https://docs.aws.amazon.com/autoscaling/ec2/userguide/using-systems-manager-parameters.html
+   *
+   */
+  public static fromSsmParameterResolve(parameterName: string, options?: SsmParameterImageOptions): IMachineImage {
```

Comment:
Maybe something like this to make it more explicit?
```suggestion
  public static resolveSsmParameterAtLaunch(parameterName: string, options?: SsmParameterImageOptions): IMachineImage {
```

Reply:
Yes I considered this option as well. Let me update it.